pkg/retainer/retainer/retainer.go

```diff
@@ -151,7 +138,7 @@ func (r *Retainer) Shutdown() {
 }
 
 func (r *Retainer) BlockMetadata(blockID iotago.BlockID) (*retainer.BlockMetadata, error) {
-	blockStatus, blockFailureReason := r.blockStatus(blockID)
+	blockStatus, blockFailureReason, _ := r.blockStatus(blockID)
```

Comment:
Get the transactionID here and make `transactionStatus` to take a transactionID instead of BlockID, that way we don't need to `getBlockData` for 2 times for the transactionID.